src/AuthorProcessor.php

```diff
@@ -0,0 +1,223 @@
+<?php 
+
+/*
+ * This file is part of the BibTex Parser. This file was contributed by Andre Chalom <andrechalom@gmail.com>.
+ *
+ * BibTex Parser is (c) Renan de Lima Barbosa <renandelima@gmail.com>
+ *
+ * This file includes source code adapted from the Structures_BibTex package, (c) Elmar Pitschke <elmar.pitschke@gmx.de>,
+ * included here under PHP license: http://www.php.net/license/3_0.txt.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace RenanBr\BibTexParser;
+
+class AuthorProcessor {
+	/** 
+	 * Entry point for this class
+	 * @param string $&value The current tag value, will be modified in-place
+	 * @param $tag The current tag. This function will only process "author" tags
+	 */
+	public function __invoke (string &$value, string $tag) {
+		if ($tag == "author") {
+			$value = $this->_extractAuthors($value);
+		}
+	}
+	/**
+	 * Unwrapping entry
+	 *
+	 * @access private
+	 * @param string $entry The entry to unwrap
+	 * @return string unwrapped entry
+	 */
+	private function _unwrap($entry)
```

Comment:
This was copied from the Structures_BibTex code. I didn't notice that in the excerpt copied this is only called once. Fixed now.

I guess there is a lot of room to improve this code - Structures_BibTex is 10 years old, after all.